### DISCUSSION TITLE
Why do longer variable names make `solveFor()` take a lot more time?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [jiggzson/nerdamer#662] [Q&A] [answered]
From #661:
> Another thing I'd like to point out is that using shorter variable names will drastically improve performance. Especially with hairier solutions such as these. Consider the two options below on the same machine.
> 
> ```js
> const equation = '((initialX1 + t * velecityX1) - (initialX2 + t * velocityX2))^2 + ((initialY1 + t * velocityY1) - (initialY2 + t * velocityY2))^2 = (radius1 + radius2) ^ 2';
> console.time('longNames')
> const preCalculated = nerdamer(equation).solveFor('t')[1];
> console.timeEnd('longNames');
> 
> // longNames: 12159.354ms
> ```
> 
> VS
> 
> ```js
> const equation = '((a+x*v1)-(b+x*v2))^2+((c+x*v3)-(d+x*v4))^2-(r1+r2)^2';
> 
> console.time('shortNames')
> const preCalculated = nerdamer(equation).solveFor('x')[1];
> console.timeEnd('shortNames')
> 
> // shortNames: 2561.224ms
> ```

Why is there around 10 second difference? Also then in this case would renaming `v1` to a single letter like `p` improve performance?

**Comment by jiggzson** [comment, marked as answer]
It seems that the issue is not related to the longer variable names but rather when solving for an equation vs an expression. The increased speed is due to the fact that solve is being provided an expression. This definitely needs investigating.
